src/app/code/local/Aligent/GeoIP/Model/Observer.php

```diff
@@ -11,22 +11,28 @@ class Aligent_GeoIP_Model_Observer {
      * on country and store id if appropriate.
      */
     public function setVarnishResponseHeader() {
-        $aHeaders = array(Aligent_GeoIP_Helper_Data::VARNISH_XGEOIP_SERVER_HEADER);
+
+        $aHeaders = array_keys(Mage::helper('aligent_geoip')->aGeoIpHeaders);
+        $oResponse = Mage::app()->getResponse();
+
 
         // If Varnish supplies a store id header, vary the response based on the store id.
         if (array_key_exists(self::VARNISH_STOREID_SERVER_VARIABLE, $_SERVER)) {
             $aHeaders[] = self::VARNISH_STOREID_HEADER;
         }
 
-        $oResponse = Mage::app()->getResponse();
         $oResponse->setHeader('Vary', implode(', ', $aHeaders) , true);
 
         // Put the request store id and country into server vars.  Useful for debugging.
         //if (array_key_exists(self::VARNISH_STORE_ID_HEADER, $_SERVER)) {
         //    $oResponse->setHeader('X-Request-StoreId', $_SERVER[self::VARNISH_STORE_ID_HEADER]);
         //}
-        if (array_key_exists(Aligent_GeoIP_Helper_Data::VARNISH_XGEOIP_SERVER_VARIABLE, $_SERVER)) {
-            $oResponse->setHeader('X-Request-GeoIP', $_SERVER[Aligent_GeoIP_Helper_Data::VARNISH_XGEOIP_SERVER_VARIABLE]);
+
+        foreach ($aHeaders as $vHeader => $vServerVariable) {
+            if (array_key_exists($vServerVariable, $_SERVER)) {
+                // Put a debugging header, so its easier to debug geo ip issue
+                $oResponse->setHeader('DEBUG-' . $vHeader, $_SERVER[$vServerVariable]);
```

Comment:
@zainengineer The only issue I have with this new version is this line...  The response header must start with `X-` for RFC compliance.  I think it should be `X-Request-` so that it's clear that we're just echoing back request data.  I'm happy if that becomes `X-Request-X-GeoIP` because we're using the existing `$vHeader` var.